plugins/api/src/touch.cpp

```diff
@@ -0,0 +1,512 @@
+#include "api_plugin_number.h"
+#include "rodsDef.h"
+#include "rcConnect.h"
+#include "rodsPackInstruct.h"
+#include "apiHandler.hpp"
+#include "client_api_whitelist.hpp"
+
+#include <functional>
+
+#ifdef RODS_SERVER
+
+//
+// Server-side Implementation
+//
+
+#include "rodsErrorTable.h"
+#include "irods_exception.hpp"
+#include "irods_server_api_call.hpp"
+#include "irods_re_serialization.hpp"
+#include "irods_logger.hpp"
+
+#define IRODS_IO_TRANSPORT_ENABLE_SERVER_SIDE_API
+#include "transport/default_transport.hpp"
+#include "dstream.hpp"
+
+#define IRODS_FILESYSTEM_ENABLE_SERVER_SIDE_API
+#include "filesystem.hpp"
+
+#define IRODS_REPLICA_ENABLE_SERVER_SIDE_API
+#include "replica.hpp"
+
+#define IRODS_QUERY_ENABLE_SERVER_SIDE_API
+#include "query_builder.hpp"
+
+#include "fmt/format.h"
+#include "json.hpp"
+
+#include <string>
+#include <string_view>
+#include <chrono>
+
+/*
+ The expected JSON format:
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
+ {
+     // Must be an absolute path.
+     // Cannot be empty.
+     "logical_path": string, 
+
+     // TODO It would be great if we can get rid of this.
+     // Doing so may require a slight change in the API.
+     //
+     // Cannot be empty.
+     "default_resource": string,
```

Comment:
yes, agreed, no default determination magic should be happening in this API call.